packages/x-data-grid/src/hooks/features/rowSelection/utils.ts

```diff
@@ -13,3 +118,137 @@ export function isMultipleRowSelectionEnabled(
   }
   return !props.disableMultipleRowSelection;
 }
+
+const getRowNodeParents = (tree: GridRowTreeConfig, id: GridRowId) => {
+  const parents: GridRowId[] = [];
+
+  let parent: GridRowId | null = id;
+
+  while (isValidRowId(parent) && parent !== GRID_ROOT_GROUP_ID) {
+    const node = tree[parent] as GridGroupNode;
+    if (!node) {
+      return parents;
+    }
+    parents.push(parent);
+    parent = node.parent;
+  }
+  return parents;
+};
+
+const getFilteredRowNodeSiblings = (
+  tree: GridRowTreeConfig,
+  filteredRows: Record<GridRowId, boolean>,
+  id: GridRowId,
+) => {
+  const node = tree[id];
+  if (!node) {
+    return [];
+  }
+
+  const parent = node.parent;
+  if (!isValidRowId(parent)) {
+    return [];
+  }
+
+  const parentNode = tree[parent] as GridGroupNode;
+
+  return [...parentNode.children].filter((childId) => childId !== id && filteredRows[childId]);
+};
+
+export const findRowsToSelect = (
+  apiRef: React.MutableRefObject<GridPrivateApiCommunity>,
+  tree: GridRowTreeConfig,
+  selectedRow: GridRowId,
+  autoSelectDescendants: boolean,
+  autoSelectParents: boolean,
+) => {
+  const filteredRows = gridFilteredRowsLookupSelector(apiRef);
+  const selectedIdsLookup = selectedIdsLookupSelector(apiRef);
+  const rowsToSelect: Set<GridRowId> = new Set([]);
+
+  if (!autoSelectDescendants && !autoSelectParents) {
+    return rowsToSelect;
+  }
+
+  if (autoSelectDescendants) {
+    const rowNode = tree[selectedRow];
+
+    if (rowNode?.type === 'group') {
+      const rowGroupDescendantsSelector = getGridRowGroupSelectableDescendantsSelector(
```

Comment:
What is the reasoning for using `createSelector` inside `getGridRowGroupSelectableDescendantsSelector`?
The selector is recreated every time `findRowsToSelect` is called, and it seems to only be called once 🤔

Reply:
> What is the reasoning for using createSelector inside getGridRowGroupSelectableDescendantsSelector?

I created these selectors when I was trying to optimize the performance as per this [proposal](https://www.notion.so/mui-org/Data-grid-row-selection-propagation-e173045a2b134abeb06cd1e9c639c0b7?pvs=4#6dde3d4aa9144f52a8c0c4285a976c31), and didn't remove them afterward, in the back of my mind I had thinking they might be required later on. Fair point though, let's make it a normal function until an explicit need for a selector arises.

> The selector is recreated every time `findRowsToSelect` is called

The need to recreate the selector with every call should be fixed once we shift to [selectors with arguments](https://github.com/mui/mui-x/pull/14236) (`createSelectorV8`) in v8, which is not possible in v7 when we need to combine the selector with the selectors created with old `createSelector` method due to the breaking change of the API.